src/Types/NonNullableType.hack

```diff
@@ -8,7 +8,7 @@ interface INonNullableType {
     public function getName(): ?string;
     public function getDescription(): ?string;
     public function getFields(bool $include_deprecated = false): ?vec<Introspection\__Field>;
-    public function getInterfaces(): ?vec<Introspection\__Type>;
+    public function getInterfaces(): ?dict<string, classname<InterfaceType>>;
```

Comment:
I just did this as a vec, I can update to a dict. I also like returning just the classname here and then having the `getIntrospectionInterfaces` deal with init'ing it 👍

Reply:
I went with the dict so that

- we can check existence by GraphQL type name in O(1)
- classnames instead of instances for performance, so that we don't construct all the instances during validation/execution